### PR TITLE
[Feature] Support refresh table for connector in new planner and analyzer

### PR DESCRIPTION
**Pull request by caneGuy**
… analyzer

## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #6509 

## Problem Summary(Required) ：

we need to support refresh_stmt on table of external catalog.
Our adaptation of external_catalog in fe only under new parser and new analyzer. So we need to support refresh_stmt in new parser and new analyzer.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 25 (04.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/RefreshTableStatement.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/RefreshTableStatementAnalyzer.java|0|11|00.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/RefreshTableStatementAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import com.starrocks.analysis.TableName;
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
+import com.starrocks.mysql.privilege.PrivPredicate;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.sql.ast.RefreshTableStatement;
+
+public class RefreshTableStatementAnalyzer {
+    public static void analyze(RefreshTableStatement statement, ConnectContext context) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/RefreshTableStatementAnalyzer.java:14]: ref CatalogAnalyzer

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 2]
```diff
@@ -126,6 +127,10 @@ showTableStatusStatement
     : SHOW TABLE STATUS ((FROM | IN) db=qualifiedName)? ((LIKE pattern=string) | (WHERE expression))?
     ;
 
+refreshTableStatement
+    : REFRESH TABLE FROM (table=qualifiedName)
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:131]:
```suggestion
    : REFRESH TABLE qualifiedName (PARTITION '(' string (',' string)* ')')?
```

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 3]
```diff
@@ -338,6 +339,13 @@ public ParseNode visitShowTableStatusStatement(StarRocksParser.ShowTableStatusSt
         return new ShowTableStatusStmt(dbName == null ? null : dbName.toString(), pattern, where);
     }
 
+    @Override
+    public ParseNode visitRefreshTableStatement(StarRocksParser.RefreshTableStatementContext context) {
+        QualifiedName qualifiedName = getQualifiedName(context.qualifiedName());
+        TableName targetTableName = qualifiedNameToTableName(qualifiedName);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:345]: Add partitionNames

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/RefreshTableStatement.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.sql.ast;
+
+import com.starrocks.analysis.DdlStmt;
+import com.starrocks.analysis.TableName;
+
+public class RefreshTableStatement extends DdlStmt {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/RefreshTableStatement.java:8]:
```suggestion
public class RefreshTableStmt extends DdlStmt {
```

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/RefreshTableStatementAnalyzer.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        TableName tableName = statement.getTbl();
+        if (tableName == null) {
+            ErrorReport.reportSemanticException(ErrorCode.ERR_NO_TABLES_USED);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/RefreshTableStatementAnalyzer.java:17]: MetaUtils.normalizationTableName(context, tableName);

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 6]
```diff
@@ -126,6 +127,10 @@ showTableStatusStatement
     : SHOW TABLE STATUS ((FROM | IN) db=qualifiedName)? ((LIKE pattern=string) | (WHERE expression))?
     ;
 
+refreshTableStatement
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:130]: add some ut

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 7]
```diff
@@ -126,6 +127,10 @@ showTableStatusStatement
...
+    : REFRESH EXTERNAL TABLE qualifiedName (PARTITION '(' string (',' string)* ')')?
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:131]:
```suggestion
    : REFRESH EXTERNAL TABLE qualifiedName partitionNames?
```

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/RefreshTableStmt.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.sql.ast;
+
+import com.starrocks.analysis.DdlStmt;
+import com.starrocks.analysis.TableName;
+
+import java.util.List;
+
+public class RefreshTableStmt extends DdlStmt {
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/RefreshTableStmt.java:10]: Please add comment explain this statement

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/RefreshTableStatementAnalyzer.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+        @Override
+        public Void visitRefreshTableStatement(RefreshTableStmt statement, ConnectContext context) {
+            TableName tableName = statement.getTbl();
+            MetaUtils.normalizationTableName(context, tableName);
+            if (tableName == null) {
+                ErrorReport.reportSemanticException(ErrorCode.ERR_NO_TABLES_USED);
+            }
+
+            if (!PrivilegeChecker.checkTblPriv(ConnectContext.get(), tableName.getCatalog(),
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/RefreshTableStatementAnalyzer.java:33]: add privilege in PrivilegeChecker.java

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 27 / 38 (71.05%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/RefreshTableStatementAnalyzer.java|9|17|52.94%|
|:large_blue_circle:|com/starrocks/sql/ast/RefreshTableStmt.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/RefreshTableStatementAnalyzer.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    static class RefreshTableStatementAnalyzerVisitor extends AstVisitor<Void, ConnectContext> {
+        public void analyze(DdlStmt statement, ConnectContext session) {
+            visit(statement, session);
+        }
+
+        @Override
+        public Void visitRefreshTableStatement(RefreshTableStmt statement, ConnectContext context) {
+            TableName tableName = statement.getTbl();
+            MetaUtils.normalizationTableName(context, tableName);
+            if (tableName == null) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/RefreshTableStatementAnalyzer.java:28]: I think you may check the table exists in here.
just call MetadataMgr.getTable() then judge it by null or not.

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/RefreshTableStatementAnalyzer.java:28]: nit : add a common method to judge the catalog/db/table exists

**Review by stephen-shelby — DISMISSED** [review]
LGTM

**Review by stephen-shelby — DISMISSED** [review]
LGTM

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Pull request merged**